src/EnergyPlus/ElectricPowerServiceManager.cc

```diff
@@ -1974,30 +1985,39 @@ GeneratorController::GeneratorController(std::string const &objectName,
         generatorType = GeneratorType::iCEngine;
         compGenTypeOf_Num = DataGlobalConstants::iGeneratorICEngine;
         compPlantTypeOf_Num = DataPlant::TypeOf_Generator_ICEngine;
+        compPlantName = name;
     } else if (UtilityRoutines::SameString(objectType, "Generator:CombustionTurbine")) {
         generatorType = GeneratorType::combTurbine;
         compGenTypeOf_Num = DataGlobalConstants::iGeneratorCombTurbine;
         compPlantTypeOf_Num = DataPlant::TypeOf_Generator_CTurbine;
+        compPlantName = name;
     } else if (UtilityRoutines::SameString(objectType, "Generator:MicroTurbine")) {
         generatorType = GeneratorType::microturbine;
         compGenTypeOf_Num = DataGlobalConstants::iGeneratorMicroturbine;
         compPlantTypeOf_Num = DataPlant::TypeOf_Generator_MicroTurbine;
+        compPlantName = name;
     } else if (UtilityRoutines::SameString(objectType, "Generator:Photovoltaic")) {
         generatorType = GeneratorType::pV;
         compGenTypeOf_Num = DataGlobalConstants::iGeneratorPV;
         compPlantTypeOf_Num = DataPlant::TypeOf_PVTSolarCollectorFlatPlate;
+        compPlantName = name;
     } else if (UtilityRoutines::SameString(objectType, "Generator:PVWatts")) {
         generatorType = GeneratorType::pvWatts;
         compGenTypeOf_Num = DataGlobalConstants::iGeneratorPVWatts;
         compPlantTypeOf_Num = DataPlant::TypeOf_Other;
     } else if (UtilityRoutines::SameString(objectType, "Generator:FuelCell")) {
         generatorType = GeneratorType::fuelCell;
         compGenTypeOf_Num = DataGlobalConstants::iGeneratorFuelCell;
-        compPlantTypeOf_Num = DataPlant::TypeOf_Generator_FCStackCooler;
+        // fuel cell has two possible plant component types, stack cooler and exhaust gas HX.
```

Comment:
Setting this up right here in the constructor, good.  I was trying to see where compPlantName was getting assigned to the member variable, but that's implied here.